### ISSUE TITLE
ENABLE_SSL_ADMIN not defined in ZC 1.5.5

### ISSUE THREAD

**Issue by Zen4All** [zencart/zencart#1376]
In admin/includes/functoins/html_output, ENABLE_SSL_ADMIN is used, but not defined.
On the front end I saw this was already changed and the define is commented out.
lines 18 through 26 are still using ENABLE_SSL_ADMIN 
```
    if ($connection == 'NONSSL') {
      $link = HTTP_SERVER . DIR_WS_ADMIN;
    } elseif ($connection == 'SSL') {
      if (ENABLE_SSL_ADMIN == 'true') {
        $link = HTTPS_SERVER . DIR_WS_HTTPS_ADMIN;
      } else {
        $link = HTTP_SERVER . DIR_WS_ADMIN;
      }
    } else {
```
and line 44 too
`      } elseif ( ( ($request_type == 'NONSSL') && ($connection == 'SSL') && (ENABLE_SSL_ADMIN == 'true') ) || ( ($request_type == 'SSL') && ($connection == 'NONSSL') ) ) {`
It is not very important,since in Zen cart 1.6 this file will change significantly, but maybe for a upcomming minor update (1.5.5e or1.5.6?) this can be changed

**Comment by drbyte** [comment]
Thanks for reporting this.

As far as core code goes, none of the code in html_output.php that references it makes any difference in this case. A NOTICE might be raised due to the constant being undefined. But none of the logic that's testing it is impacted. 

The only downside as far as functionality goes, is that the session cookie wouldn't be set as https-only. To fix that issue, simply edit the `admin` `init_sessions.php` and change:

```diff
-$secureFlag = ((ENABLE_SSL_ADMIN == 'true' && substr(HTTP_SERVER, 0, 6) == 'https:' && substr(HTTPS_SERVER, 0, 6) == 'https:') || (ENABLE_SSL_ADMIN == 'false' && substr(HTTP_SERVER, 0, 6) == 'https:')) ? TRUE : FALSE;
+$secureFlag = (substr(HTTP_SERVER, 0, 6) == 'https:') ? TRUE : FALSE;
```

**Comment by drbyte** [comment]
Closed by #1394
